client/src/app/core/ui-services/tree.service.ts

```diff
@@ -218,43 +218,130 @@ export class TreeService {
+                results.push(branch);
+            } else if (branch.children && branch.children.length) {
+                results = results.concat(this.getBranchesFromTree(branch.children, items));
             }
         });
         return results;
     }
 
     /**
-     * Return all items not being hierarchically dependant on the items in the input arrray
+     * Inserts OSTreeNode branches into another tree at the position specified
+     *
+     * @param tree A (partial) tree the branches need to be inserted into. It
+     * is assumed that this tree does not contain the branches to be inserted.
+     * See also {@link getTreeWithoutSelection}
+     * @param branches OsTreeNodes to be inserted. See also {@link getBranchesFromTree}
+     * @param parentId the id of a parent node under which the branches should be inserted
+     * @param olderSibling (optional) the id of the item on the same level
+     * the tree is to be inserted behind
+     * @returns the re-arranged tree containing the branches
+     */
+    public insertBranchesIntoTree<T extends Identifiable & Displayable>(
+        tree: OSTreeNode<T>[],
+        branches: OSTreeNode<T>[],
+        parentId: number,
+        olderSibling?: number
+    ): OSTreeNode<T>[] {
+        if (!parentId && olderSibling) {
+            const older = tree.findIndex(branch => branch.id === olderSibling);
+            if (older >= 0) {
+                return [...tree.slice(0, older + 1), ...branches, ...tree.slice(older + 1)];
+            }
+        } else if (parentId) {
+            for (const branch of tree) {
+                if (branch.id !== parentId) {
+                    if (branch.children && branch.children.length) {
+                        branch.children = this.insertBranchesIntoTree(
+                            branch.children,
+                            branches,
+                            parentId,
+                            olderSibling
+                        );
+                    }
+                } else {
+                    if (!branch.children) {
+                        branch.children = branches;
+                    } else {
+                        if (olderSibling) {
+                            const older = branch.children.findIndex(child => child.id === olderSibling);
+                            if (older >= 0) {
+                                branch.children = [
+                                    ...branch.children.slice(0, older + 1),
+                                    ...branches,
+                                    ...branch.children.slice(older + 1)
+                                ];
+                            }
+                        } else {
+                            branch.children = [...branch.children, ...branches];
+                        }
+                    }
+                }
+            }
+            return tree;
+        } else {
+            throw new Error('This should not happen. Invalid sorting items given');
```

Comment:
This happens, if you e.g. select every motion and select something in the (empty) dialog. Maybe let the user do not open the dialog, if there is nothing, he can select...

Reply:
I will throw an error (resulting in a error message: 'No items left o chose from')